src/test/scala/mesosphere/marathon/integration/setup/MesosTest.scala

```diff
@@ -335,37 +335,18 @@ case class MesosCluster(
 trait MesosTest {
   def mesos: MesosFacade
   val mesosMasterUrl: String
-  def cleanMesos(): Unit
 }
 
-object MesosTest {
-  def clean(client: MesosFacade, cleanTimeout: Duration = 5.minutes)(implicit ec: ExecutionContext, s: Scheduler): Unit = {
-    def teardown: Future[Done] =
-      Retry("teardown marathon", maxDuration = cleanTimeout, minDelay = 0.25.second, maxDelay = 2.second) {
-        Future.foldLeft(client.frameworkIds().value.map(client.teardown(_).map { response =>
-          val status = response.status.intValue
-          if (status == 200) Done
-          else throw new IllegalStateException(s"server returned status $status")
-        }))(Done) { (_, _) => Done }.map { _ =>
-          val agents = client.state.value.agents
-          agents.find(a => !a.usedResources.isEmpty || a.reservedResourcesByRole.nonEmpty).fold(Done) { agent =>
-            throw new IllegalStateException(
-              s"agent allocated on agent ${agent.id}: " +
-                s"used = ${agent.usedResources}, reserved = ${agent.reservedResourcesByRole}")
-          }
-        }
-      }
-    Await.result(teardown, Duration.Inf)
-  }
-}
+object MesosTest extends Eventually {
+  def teardown(facade: MesosFacade): Unit = {
```

Comment:
Do we use this only in `trait Mesos`? Cause then we could move the functionality.

Reply:
You're right. No reason to keep this object around.